src/interface.ts

```diff
@@ -80,9 +89,9 @@ interface DesktopAgent {
   broadcast(context: Context): void;
 
   /**
-   * Constructs a new intent
+   * Raises an intent to the desktop agent to resolve.
    */
-  intent(intent: IntentName, context: Context, target: String): Intent;
+  raiseIntent(intent: IntentName, context: Context, target: String): Promise<IntentResolution>;
```

Comment:
:+1: 

I would change it to the following however:
`raiseIntent(intent: IntentName, context: Context, target?: AppIdentifier): Promise<IntentResolution>;`

`target` should definitely be optional, since it is not always known for sending an intent.